agent/envoy_bootstrap/envoy_bootstrap.go

```diff
@@ -286,23 +301,25 @@ func getRegionalXdsEndpoint(region string, envoyCLIInst EnvoyCLI) (*string, erro
 	if err != nil {
 		return nil, err
 	}
-
-	version, err := envoyCLIInst.run("--version")
-	if err != nil {
-		log.Warnf("Could not determine envoy version: %v", err)
-		version = "unknown"
+	var useFips bool
+	if fips, err := isFipsCompatible(region, envoyCLIInst); err != nil {
+		return nil, err
+	} else {
+		// TODO: In GovCloud regions the AppMesh FIPS xDS endpoint does not contain the suffix `-fips`.
+		// In future if AppMesh adds endpoints with suffx '-fips' in GovCloud regions then get rid of
+		// this below logic and directly use the variable `fips`.
+		useFips = fips && !strings.HasPrefix(region, "us-gov-")
```

Comment:
Nit: can we add the prefix check for gov cloud in part of the `isFipsCompatible` function? I know this is temporary so totally fine if we skip.

Reply:
The intention here is that in future we just get rid of the part "`!strings.HasPrefix(region, "us-gov-")`"  when AppMesh xDS in GovCloud supports endpoints with `-fips` suffix.

I originally had the approach you mentioned, but now that in subsequent commits we are reusing the same function for relay mode, I thought of keeping the `isFipsCompatible` function simple & generic.